File: c++/Singleton/Singleton.cpp
#include "Singleton.hpp"

Singleton* Singleton::_INSTANCE = nullptr;

Singleton* Singleton::getInstance() {

    if (!_INSTANCE) {
        _INSTANCE = new Singleton();
    }

    return _INSTANCE;
}

void Singleton::setNumber(int number) {
    _number = number;
}

int Singleton::getNumber() {
    return _number;
}

File: java/Singleton/Singleton.java

public class Singleton {

    private int _number;

    static Singleton _INSTANCE;

    private Singleton() {

    }

    public static Singleton getInstance() {
        if (_INSTANCE == null) {
            _INSTANCE = new Singleton();
        }
        
        return _INSTANCE;
    }

    public void setNumber(int number) {
        this._number = number;
    }

    public int getNumber() {
        return this._number;
    }
    
}